### PR TITLE
Added documentation for CanonicalTransformation and CanonicalBlocks

### PR DESCRIPTION
**Pull request by sriubas**
Added documentation for functions `CanonicalTransformation` and `CanonicalBlocks`. Also removed an unused variable from the code of `CanonicalBlocks`.

### CODE REVIEW

File: doc/blocks.xml [hunk 1]
```diff
@@ -173,3 +173,27 @@ gap> DegreeOfBlocks(blocks);
...
   </ManSection>
 <#/GAPDoc>
+
+<#GAPDoc Label="CanonicalBlocks">
+  <ManSection>
+    <Attr Name = "CanonicalBlocks" Arg = "blocks"/>
+    <Returns>Blocks of a bipartition.</Returns>
+    <Description>
+      For the blocks of a bipartition <A>blocks</A> the function
+      <C>CanonicalBlocks</C> returns a canonical represnetative of the
```
Comment (wilfwilson) [inline doc/blocks.xml]: wrong spelling - should be "representative"

File: doc/blocks.xml [hunk 2]
```diff
@@ -173,3 +173,27 @@ gap> DegreeOfBlocks(blocks);
...
+      <A>blocks</A>.
+      <P/>
+
+      In particular, let <C>C</C> be a class of blocks such that for every pair
+      of elements <C>x</C> and <C>y</C> of <C>C</C> the number of signed, and
+      similarly unsigned, blocks of any given size in both <C>x</C> and
+      <C>y</C> are the same. Then <C>CanonicalTransformation</C> returns a
+      canonical represnetative of <C>C</C>.
```
Comment (wilfwilson) [inline doc/blocks.xml]: Same comment about the spelling of representative.

File: doc/blocks.xml (hunk 1 above)
Comment (wilfwilson) [inline doc/blocks.xml]: Also say "of <A>blocks</A>" - i.e. delete "the"

File: doc/blocks.xml [hunk 3]
```diff
@@ -173,3 +173,27 @@ gap> DegreeOfBlocks(blocks);
     </Description>
   </ManSection>
 <#/GAPDoc>
+
+<#GAPDoc Label="CanonicalBlocks">
+  <ManSection>
+    <Attr Name = "CanonicalBlocks" Arg = "blocks"/>
+    <Returns>Blocks of a bipartition.</Returns>
+    <Description>
+      For the blocks of a bipartition <A>blocks</A> the function
```
Comment (wilfwilson) [inline doc/blocks.xml]: When I read this, it made it seem to me that the argument is a bipartition. Maybe reword this
"If <A>blocks</A> is the blocks of a bipartition, then ..."

File: doc/blocks.xml [hunk 4]
```diff
@@ -173,3 +173,27 @@ gap> DegreeOfBlocks(blocks);
...
+      <C>CanonicalBlocks</C> returns a canonical represnetative of the
+      <A>blocks</A>.
+      <P/>
+
+      In particular, let <C>C</C> be a class of blocks such that for every pair
+      of elements <C>x</C> and <C>y</C> of <C>C</C> the number of signed, and
+      similarly unsigned, blocks of any given size in both <C>x</C> and
+      <C>y</C> are the same. Then <C>CanonicalTransformation</C> returns a
```
Comment (wilfwilson) [inline doc/blocks.xml]: Should be blocks rather than transformation.

File: doc/blocks.xml [hunk 5]
```diff
@@ -173,3 +173,27 @@ gap> DegreeOfBlocks(blocks);
...
+  <ManSection>
+    <Attr Name = "CanonicalBlocks" Arg = "blocks"/>
+    <Returns>Blocks of a bipartition.</Returns>
+    <Description>
+      For the blocks of a bipartition <A>blocks</A> the function
+      <C>CanonicalBlocks</C> returns a canonical represnetative of the
+      <A>blocks</A>.
+      <P/>
+
+      In particular, let <C>C</C> be a class of blocks such that for every pair
```
Comment (wilfwilson) [inline doc/blocks.xml]: I must say I don't understand this paragraph.

File: doc/trans.xml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  <ManSection>
+    <Func Name = "CanonicalTransformation" Arg = "trans [,n]"/>
+    <Returns>
+      A transformation.
+    </Returns>
+    <Description>
+      If <A>trans</A> is a transformation, and <A>n</A> is a non-negative
+      integer such that the restriction of <A>trans</A> to <K>[1 .. n]</K>
+      defines a transformation of <C>[1 .. n]</C>, then
+      <C>CanonicalTransformation</C> returns a canonical represnetation of the
```
Comment (wilfwilson) [inline doc/trans.xml]: representative, rather than represnetation

File: doc/trans.xml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      transformation <A>trans</A> restricted to <C>[1 .. n]</C>. 
+      <P/>
+
+      More specifilly, let <C>C</C> be a class of transformations of degree
```
Comment (wilfwilson) [inline doc/trans.xml]: Misspelling - should be "specifically"

File: doc/trans.xml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      <A>n</A> such that <C>AsDigraph</C>  return isomorphic digraphs for every
+      pair of element elements in <C>C</C>. Recall that for a transformation
+      <A>trans</A> and integer <A>n</A> the function <C>AsDigraph</C> returns a
+      digraph with <A>n</A> vertices and an edge with source <C>x</C> and range
+      <C>x^trans</C> for every <C>x</C> in <C>[1 .. n]</C>. See <Ref
+      Oper="AsDigraph" BookName="Digraphs"/>. Then
+      <C>CanonicalTransformation</C> returns a canonical represnetative of
```
Comment (wilfwilson) [inline doc/trans.xml]: representative

File: doc/trans.xml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+      <C>C</C>.
+
```
Comment (wilfwilson) [inline doc/trans.xml]: Maybe have a sentence saying that therefore canonicaltransformation returns the canonical representative of the class of transformations to which the argument belongs. To be super explicit.

File: doc/blocks.xml (hunk 5 above)
Comment (wilfwilson) [inline doc/blocks.xml]: Ah. I was reading this to mean that `C` was on of the parts of the blocks <A>blocks</A>.

File: doc/blocks.xml (hunk 2 above)
Comment (wilfwilson) [inline doc/blocks.xml]: As I said below on the doc about transformations, perhaps end with a statement declaring precisely what the function does to the specific argument. i.e. returns the canonical rep of the class to which the arg belongs.

**Comment by sriubas** [comment]
I think, I've made all the changes now.

**Comment by wilfwilson** [comment]
Thanks, I'll have a quick look

File: doc/trans.xml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    <Description>
+      If <A>trans</A> is a transformation, and <A>n</A> is a non-negative
+      integer such that the restriction of <A>trans</A> to <K>[1 .. n]</K>
+      defines a transformation of <C>[1 .. n]</C>, then
+      <C>CanonicalTransformation</C> returns a canonical representative of the
+      transformation <A>trans</A> restricted to <C>[1 .. n]</C>. 
+      <P/>
+
+      More specifically, let <C>C(n)</C> be a class of transformations of degree
+      <A>n</A> such that <C>AsDigraph</C>  return isomorphic digraphs for every
```
Comment (wilfwilson) [inline doc/trans.xml]: return -> returns

File: doc/blocks.xml [hunk 11]
```diff
@@ -173,3 +173,29 @@ gap> DegreeOfBlocks(blocks);
     </Description>
   </ManSection>
 <#/GAPDoc>
+
+<#GAPDoc Label="CanonicalBlocks">
+  <ManSection>
+    <Attr Name = "CanonicalBlocks" Arg = "blocks"/>
+    <Returns>Blocks of a bipartition.</Returns>
+    <Description>
+      If <A>blocks</A> is blocks of a bipartition, then the function
```
Comment (wilfwilson) [inline doc/blocks.xml]: "blocks" -> "the blocks"

File: doc/trans.xml [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+      pair of element elements in <C>C(n)</C>. Recall that for a transformation
+      <A>trans</A> and integer <A>n</A> the function <C>AsDigraph</C> returns a
+      digraph with <A>n</A> vertices and an edge with source <C>x</C> and range
+      <C>x^trans</C> for every <C>x</C> in <C>[1 .. n]</C>. See <Ref
+      Oper="AsDigraph" BookName="Digraphs"/>. Then
+      <C>CanonicalTransformation</C> returns a canonical representative of
+      <C>C(n)</C> containing <A>trans</A>.
```
Comment (wilfwilson) [inline doc/trans.xml]: I'd recommend `<C>C(n)</C> containing <A>trans</A>` -> `the class <C>C(n)</C> that contains <A>trans</A>`

**Comment by sriubas** [comment]
Done
